alica_engine/include/engine/scheduler/Job.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include <iostream>
+#include <chrono>
+#include <ctime>
+#include <vector>
+#include <memory>
+
+namespace alica {
+namespace scheduler
+{
+struct Job
```

Comment:
Add a constructor that will take cb function and preRequisites as mandatory arguments, scheduledTime as optional argument with default value as clock.now().  Internally set cancelled, isRepeated to false.